src/configupdater/parser.py

```diff
@@ -455,7 +457,8 @@ def _read(self, fp: Iterable[str], fpname: str, into: Document):
                         and optname
                         and cursect[optname] is not None
                     ):
-                        cursect[optname].append("")  # newlines added at join
+                        # newlines added at join
+                        cast(List, cursect[optname]).append("")
```

Comment:
So `mypy` is ignoring the condition `and cursect[optname] is not None`?
Does it complain if the cast is not present?

Reply:
Yes, exactly. Normally `mypy` should be able to infer that if `cursect[optname]` is `Optional[List[...]] `then it has to be `List[...]` after the if statement. Somehow it doesn't and just complains here if I don't add the cast.